src/plugins/data/common/datasets/types.ts

```diff
@@ -256,6 +258,29 @@ export interface DatasetField {
   // TODO:  osdFieldType?
 }
 
+export enum DatasetCanvasBannerComponentType {
+  CALLOUT = 'callout',
+  CUSTOM = 'custom',
+}
+
+export type DatasetCanvasBannerProps =
+  | {
+      componentType: DatasetCanvasBannerComponentType.CALLOUT;
```

Comment:
if my understanding of this is correct. This should just be `type` as it is redundant if this component is already a component and developers will understand that the `type` is associated to the type of component.